python/cudf/cudf/_lib/copying.pyx

```diff
@@ -704,3 +706,19 @@ def sample(Table input, size_type n,
             else input._index_names
         )
     )
+
+def segmented_gather(Column source_column, Column gather_map):
+    cdef shared_ptr[lists_column_view] source_LCV = (
+        make_shared[lists_column_view](source_column.view())
+    )
+    cdef shared_ptr[lists_column_view] gather_map_LCV = (
+        make_shared[lists_column_view](gather_map.view())
+    )
```

Comment:
Can we create these on the stack instead of the heap? Their scope is within this function.

Reply:
This seems to get in the way..
https://github.com/rapidsai/cudf/blob/04f90211fe95b99f4ad934f273faf87639b1f6ff/python/cudf/cudf/_lib/types.pyx#L172-L175

Reply:
Yeah without a default constructor, we need to construct these on the heap. At least wrapping it in a `shared_ptr` doesn't need us to manually `del` the object